treeherder/model/bug_suggestions.py

```diff
@@ -0,0 +1,231 @@
+import logging
+import re
+import urllib
+import json
+
+from django.core.urlresolvers import reverse
+from django.conf import settings
+
+from thclient import TreeherderArtifactCollection
+from treeherder.etl import th_publisher
+
+logger = logging.getLogger(__name__)
+
+
```

Comment:
From here to the next point, it's mostly new code handling the new artifacts and the posting, etc.